lib/segment_tree.rb

```diff
@@ -84,17 +78,9 @@ def inspect
   end
 
   private
-  def matches?(x, low_idx, high_idx, idx) #:nodoc:
-    low, high = @segments[low_idx], @segments[high_idx]
-    segment   = @segments[idx]
-    left      = idx > 0 && @segments[idx - 1]
-    right     = idx < @segments.size - 1 && @segments[idx + 1]
-
-    case
-      when left && low.range.begin <= x && x <= left.range.end then -1
-      when segment.range.begin <=x && x <= segment.range.end then 0
-      when right && right.range.begin <=x && x <= high.range.end then 1
-      else nil
-    end
+  def matches?(x, low_idx, idx, high_idx) #:nodoc:
```

Comment:
I don't really like how code looks now. That variables you've removed were mostly for self-documenting. Now code is harder to understand